### ISSUE TITLE
User Words: Case & Plurals

### ISSUE THREAD

**Issue by erosman** [streetsidesoftware/vscode-spell-checker#2117]
Are the User Words case-sensitive?
Are they English plural aware?

For example:
userscript, userscripts, UserScript, UserScripts, userScript, userScripts

Should all above be entered? or is there flag for case sensitive/insensitive?

**Comment by Jason3S** [comment]
@erosman,

I'll try and answer your questions:

Reference: [Case Sensitivity - CSpell](https://cspell.org/docs/case-sensitive/) and 

- Dictionaries are case sensitive.
- English stemming is NOT applied - auto stemming is great for searching, but it isn't great for spell checking.
- There is a setting for case sensitivity: `cSpell.caseSensitive`
   ![image](https://user-images.githubusercontent.com/3740137/183235590-4495ddba-b409-428d-97b6-f7597f041741.png)

## Dictionaries

The words in a dictionary are case sensitive. When a custom dictionary is loaded, a non-case sensitive version is made.

My suggestion is to add words in the form you expect.

**Comment by erosman** [comment]
> The words in a dictionary are case sensitive.

That is expected and I wouldn't change it.

> When a custom dictionary is loaded, a non-case sensitive version is made.

Is user words a custom dictionary?

With regards to User Words (only), having a flag could be helpful as case-insensitive User Words is not optimal either.

e.g.: userscript/i

**Comment by Jason3S** [comment]
@erosman,

User Words / Words are treated as a custom dictionary. Only they are "read" from the configuration instead of a file.

You can also forbid words by prefixing it with `!`:
```js
"cSpell.userWords": ["Userscript", "!userscript"] // `Userscript` is ok, but `userscript` marked as incorrect. 
```

**Comment by erosman** [comment]
Thank you :+1:
